packages/contracts-periphery/contracts/foundry-tests/Optimist.t.sol

```diff
@@ -0,0 +1,402 @@
+            key: bytes32("optimist.can-mint"),
+            val: bytes("true")
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
+        // mint as bob
+        vm.prank(bob);
+        optimist.mint(bob);
+
+        // attempt to approve sally
+        vm.prank(bob);
+        vm.expectRevert("Optimist: soul bound token");
+        optimist.approve(address(attestationStation), 256);
+
+        assertEq(optimist.getApproved(256), address(0));
+    }
+
+    /**
+     * @dev It should be able to burn token
+     */
+    function test_optimist_burn() external {
+        AttestationStation attestationStation = new AttestationStation();
+        Optimist optimist = new Optimist(name, symbol, alice_admin, attestationStation);
+
+        // whitelist bob
+        AttestationStation.AttestationData[]
+            memory attestationData = new AttestationStation.AttestationData[](1);
+        // we are using true but it can be any non null value
+        attestationData[0] = AttestationStation.AttestationData({
+            about: bob,
+            key: bytes32("optimist.can-mint"),
+            val: bytes("true")
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
+        // mint as bob
+        vm.prank(bob);
+        optimist.mint(bob);
+
+        // burn as bob
+        vm.prank(bob);
+        optimist.burn(256);
+
+        // expect bob to have no balance now
+        assertEq(optimist.balanceOf(bob), 0);
+    }
+
+    /**
+     * @dev setApprovalForAll should revert as sbt
+     */
+    function test_optimist_set_approval_for_all() external {
+        AttestationStation attestationStation = new AttestationStation();
+        Optimist optimist = new Optimist(name, symbol, alice_admin, attestationStation);
+
+        // whitelist bob
+        AttestationStation.AttestationData[]
+            memory attestationData = new AttestationStation.AttestationData[](1);
+        // we are using true but it can be any non null value
+        attestationData[0] = AttestationStation.AttestationData({
+            about: bob,
+            key: bytes32("optimist.can-mint"),
+            val: bytes("true")
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
+        // mint as bob
+        vm.prank(bob);
+        optimist.mint(bob);
+        vm.prank(alice_admin);
+        vm.expectRevert(bytes("Optimist: soul bound token"));
+        optimist.setApprovalForAll(alice_admin, true);
+
+        // expect approval amount to stil be 0
+        assertEq(optimist.getApproved(256), address(0));
+        // isApprovedForAll should return false
+        assertEq(optimist.isApprovedForAll(alice_admin, alice_admin), false);
+    }
+
+    /**
+     * @dev should support erc721 interface
+     */
+    function test_optimist_supports_interface() external {
+        AttestationStation attestationStation = new AttestationStation();
```

Comment:
You could put the deployment of the `AttestationStation` and the `Optimist` in the `setUp` function